tests/cnf/core/network/metallb/internal/frr/frr.go

```diff
@@ -230,6 +269,60 @@ func DefineBGPConfigWithStaticRouteAndNetwork(localBGPASN, remoteBGPASN int, hub
 	return bgpConfig
 }
 
+// DefineBGPConfigWithUnnumbered defines BGP config file with static route and network.
+func DefineBGPConfigWithUnnumbered(localBGPASN, remoteBGPASN int, interfaceName, peerLinkLocalAddress string,
+	advertisedIPv4Routes, advertisedIPv6Routes []string, multiHop, bfd bool) string {
+	bgpConfig := tsparams.FRRBaseConfig +
+		fmt.Sprintf("interface %s", interfaceName) +
+		"\n ipv6 nd ra-interval 10" +
+		"\n no ipv6 nd suppress-ra" +
+		"\n exit\n!\n" +
+		fmt.Sprintf("router bgp %d\n", localBGPASN) +
+		tsparams.FRRDefaultBGPPreConfig
+	bgpConfig += "!\n  neighbor unnumbered peer-group\n"
+	bgpConfig += fmt.Sprintf("  neighbor %s interface peer-group unnumbered\n", interfaceName)
+
+	if bfd {
+		bgpConfig += "  neighbor unnumbered bfd\n"
+	}
+
+	if multiHop {
+		bgpConfig += "  neighbor unnumbered ebgp-multihop 2\n"
+	}
+
+	bgpConfig += fmt.Sprintf("  neighbor unnumbered remote-as %d\n", remoteBGPASN)
+	bgpConfig += fmt.Sprintf("  neighbor unnumbered password %s\n", tsparams.BGPPassword)
+	bgpConfig += "  neighbor unnumbered timers 30 90\n  neighbor unnumbered bfd\n"
+
+	bgpConfig += "!\naddress-family ipv4 unicast\n"
+	bgpConfig += "  neighbor unnumbered activate\n"
+
+	for _, route := range advertisedIPv4Routes {
+		bgpConfig += fmt.Sprintf("  network %s\n", route)
+	}
+
+	bgpConfig += "exit-address-family\n"
+
+	// Add network commands only once for IPv6
+	bgpConfig += "!\naddress-family ipv6 unicast\n"
+	bgpConfig += "  neighbor unnumbered activate\n"
+
+	for _, route := range advertisedIPv6Routes {
+		bgpConfig += fmt.Sprintf("  network %s\n", route)
+	}
+
+	bgpConfig += "exit-address-family\n"
+	bgpConfig += "!\nroute-map RMAP permit 10"
```

Comment:
Is this ! comment or ?

Reply:
Its a line with "!" in the frr config file.

Reply:
So this is  a comment and can be removed, no?

Reply:
It looks like this 
    !
    address-family ipv4 unicast
      neighbor unnumbered activate
    exit-address-family
    !
    route-map RMAP permit 10
    set ipv6 next-hop prefer-global 
    !